recipes/install-elasticsearch.rb

```diff
@@ -0,0 +1,105 @@
+# Cookbook Name:: mh-opsworks-recipes
+# Recipe:: install-elasticsearch
+
+::Chef::Recipe.send(:include, MhOpsworksRecipes::RecipeHelpers)
+::Chef::Recipe.send(:include, MhOpsworksRecipes::DeployHelpers)
+
+elk_info = get_elk_info
+
+es_major_version = elk_info[:es_major_version]
+es_version = elk_info[:es_version]
+es_cluster_name = elk_info[:es_cluster_name]
+data_path = elk_info[:es_data_path]
+es_heap_size = xmx_ram_for_this_node(0.5)
+
+apt_repository 'elasticsearch' do
+  uri "http://packages.elasticsearch.org/elasticsearch/#{es_major_version}/debian"
+  components ['stable', 'main']
+  keyserver 'ha.pool.sks-keyservers.net'
+  key '46095ACC8548582C1A2699A9D27D666CD88E42B4'
+end
+
+include_recipe "mh-opsworks-recipes::update-package-repo"
+install_package("elasticsearch=#{es_version}")
+
+service 'elasticsearch' do
+  supports :restart => true
+  action :enable
+end
+
+execute "install kopf plugin" do
+  not_if { ::Dir.exist?("/usr/share/elasticsearch/plugins/kopf") }
+  command '/usr/share/elasticsearch/bin/plugin install lmenezes/elasticsearch-kopf/2.0'
+  timeout 30
+  retries 5
+  retry_delay 10
+end
+
+cookbook_file "kopf_settings" do
+  path '/usr/share/elasticsearch/plugins/kopf/_site/kopf_external_settings.json'
+  source "kopf_external_settings.json"
+  owner 'root'
+  group 'root'
+  mode '644'
+end
+
+template '/etc/default/elasticsearch' do
+  source 'elasticsearch-default.erb'
+  owner 'root'
+  group 'root'
+  mode '644'
+  variables({
+    heap_size: es_heap_size
+  })
+end
+
+directory "/vol/data" do
+  owner 'elasticsearch'
+  group 'elasticsearch'
+  mode '755'
+  recursive true
+end
+
+template '/etc/elasticsearch/elasticsearch.yml' do
+  source 'elasticsearch.yml.erb'
+  owner 'root'
+  group 'root'
+  mode '644'
+  variables({
+    cluster_name: es_cluster_name,
+    data_path: data_path
+  })
+  notifies :restart, "service[elasticsearch]", :immediately
+end
+
+directory "template_dir" do
+  path "/etc/elasticsearch/templates"  
+  owner 'root'
+  group 'root'
+  mode '755'
+end
+
+cookbook_file "useractions_template" do
+  path "/etc/elasticsearch/templates/useractions.json"
+  source "useractions.json"
+  owner 'root'
+  group 'root'
+  mode '644'
+end
+
+http_request "delete_templates" do
```

Comment:
The existing index data will not be affected by having the mapping deleted and replaced. The mapping is applied to incoming documents at index time.

I'm taking this out anyway as I realize my intentions here were half-baked. Any modification of existing templates needs to be done much more carefully.